### PR TITLE
fix: stabilize weekly KR and IN reference refresh

### PR DESCRIPTION
**Pull request by xang1234**
## Summary
- Add a KR KRX fallback from the last-completed trading-day listing snapshot to current listings when the historical fetch has no supported KOSPI/KOSDAQ rows.
- Preserve explicit KR metadata for requested historical date, fallback mode, fallback row counts, and baseline drift warnings.
- Add weekly Asia refresh progress output during yfinance fundamentals hydration and reduce non-US weekly yfinance per-ticker delay.
- Add focused tests for KR fallback behavior, empty fallback failure, IN official-source merge behavior, weekly script wiring, and yfinance progress reporting.

## Root Cause
- KR weekly refresh failed deterministically because KRX historical ticker calls for 2026-04-30 returned no usable KOSPI/KOSDAQ rows and the official universe path hard-failed instead of falling back to current listings.
- IN did not show an app exception in Actions; the hosted runner lost communication after a long weekly bundle build, likely during non-US yfinance fundamentals hydration.

## Validation
- `PYTHONPATH=. /Users/admin/StockScreenClaude/backend/venv/bin/python -m pytest tests/unit/test_official_market_universe_source_service.py tests/unit/test_kr_market_data_service.py tests/unit/test_weekly_reference_scripts.py tests/unit/test_hybrid_fundamentals_service_policy.py -q` -> 51 passed.
- `git diff --check` passed.
- KR source smoke returned current fallback metadata with 2703 rows: 924 KOSPI and 1779 KOSDAQ.

## Notes
- Full backend `pytest` still has unrelated pre-existing failures, including localhost-dependent integration collection and unrelated unit failures.

## Summary by CodeRabbit

* **New Features**
  * Added real-time progress reporting during fundamentals batch fetch operations, enabling visibility into data processing stages.

* **Bug Fixes**
  * Enhanced Korea stock exchange (KRX) snapshot data retrieval with a two-stage fallback strategy. When historical listing data is unavailable, the system automatically falls back to current listings, improving data completeness and reliability.

### CODE REVIEW

File: backend/app/services/official_market_universe_source_service.py [hunk 1]
```diff
@@ -190,11 +190,25 @@ def fetch_jp_snapshot(self) -> OfficialMarketUniverseSnapshot:
...
+            historical_listing_empty = True
+            logger.warning(
+                "KR official universe fetch returned no rows for historical as_of=%s; "
+                "retrying current KRX listing finder.",
+                requested_listing_as_of.isoformat(),
+            )
+            rows = list(provider.listing_rows(boards=("KOSPI", "KOSDAQ"), as_of=None))
+            if self._kr_kospi_kosdaq_row_count(rows) > 0:
+                krx_listing_mode = "current_listing_fallback"
+                listing_as_of = self._seoul_today()
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/app/services/official_market_universe_source_service.py]: Addressed in `backend/app/services/official_market_universe_source_service.py`: the current-listing fallback now keeps the calendar-derived `requested_listing_as_of` instead of Seoul calendar today, so weekend/holiday dates cannot leak into `snapshot_as_of` or `snapshot_id`. Updated `test_fetch_kr_snapshot_falls_back_to_current_listing_rows_when_historical_empty` to cover a Saturday Seoul date.

**Comment by xang1234** [comment]
Addressed the remaining CodeRabbit outside-diff feedback as well:

- `backend/app/services/official_market_universe_source_service.py`: filters unsupported KR boards before enrichment/counting so KONEX rows cannot appear in `snapshot.rows` or `source_count` when metadata says they are excluded.
- `backend/app/services/hybrid_fundamentals_service.py`: phase-1 progress checkpoint is now an integer and never drops to zero for non-empty batches.
- `backend/app/services/bulk_data_fetcher.py`: progress callback failures are logged and ignored so telemetry failures cannot abort fundamentals hydration.

Validation: focused branch suite now passes with 63 tests.
